cpp/util/etcd.cc

```diff
@@ -688,14 +726,24 @@ void EtcdClient::FetchDone(RequestState* etcd_req, Task* task) {
   VLOG(2) << "EtcdClient::FetchDone: " << task->status();
 
   if (!task->status().ok()) {
-    // TODO(pphaneuf): If there is a connection problem, we should re-do the
-    // request on another node.
+    if (task->status().error_code() == util::error::UNAVAILABLE) {
+      // Seems etcd wasn't available; pick a new etcd server and retry
+      LOG(WARNING) << "Etcd fetch failed: " << task->status() << ", retrying "
+                   << "on next etcd server.";
+      ChooseNextServer();
```

Comment:
Why call this here? If you keep `endpoint_`, you could just call `UpdateEndpoint`, but I take `endpoint_` out entirely, and use `etcds_.front()` instead...

Note that ripping it out will properly find everything that's using it, as a side-effect, so would catch little details like `etcd_req->SetHostPort(endpoint_);` without having updated `etcds_`...